src/qibocal/auto/task.py

```diff
@@ -134,6 +135,18 @@ def run(
         completed = Completed(self, Normal(), folder)
         task_qubits = self._allocate_local_qubits(qubits, platform)
 
+        # get parent and child class data_members
+        parameters_keys = set(
+            self.operation.parameters_type.__dict__["__annotations__"].keys()
+        ) | set(Parameters.__dict__["__annotations__"].keys())
+        # read all settings defined in platform-specific qibolab runcard
+        for setting, value in platform.settings.__dict__.items():
+            # save values non defined in qibocal runcard,
+            # but that are used by the class
+            if setting not in self.action.parameters:
+                if setting in parameters_keys:
+                    self.action.parameters[setting] = value
```

Comment:
```suggestion
for setting in parameters_keys:
            if setting not in self.action.parameters:
                if hasattr(platform.settings, setting):
                    self.action.parameters[setting]=getattr(platform.settings, setting)

```
I like your solution. I understand that for a few parameters, maybe the @andrea-pasquale approach is better, but this is more general. I suggest not using the dictionaries for better readability.

Reply:
I know that maybe it is too late but if @andrea-pasquale agrees I prefer to restore your solution, in this way it is easier to add default parameters. With my suggested changes (they should work) the only convoluted part is the `parameters_keys` definition, I didn't find a better solution for now.

Reply:
I agree. I'll commit your suggestion as soon as he replies.

Reply:
I would go with @andrea-pasquale solution as it is the one I already implemented in qibolab and I suggested the nshots and relaxation_time as parameters for all routines and I could thing of any other that needs to be shared.

Reply:
To be more specific this approach will only work with attributes in `platform.settings` which are only 3 at the moment:
https://github.com/qiboteam/qibolab/blob/53737e12aa7400b22e100eabf538fab9a77667d6/src/qibolab/platform.py#L23-L32
And the sampling rate should not be there I believe.

Reply:
Also, maybe before discussing which one to use we should get one that passes tests, right ?

Reply:
Ok, @andrea-pasquale's solution doesn't work. @Edoardo-Pedicillo's and mine work.
The problem is that `self.parameters()` calls `<SomeChildParameterClass>.load()` which pops elements from its input argument.

I think we should definitely avoid changing the value of function arguments, e.g.:
```
    @classmethod
    def load(cls, input_parameters):
        """Load parameters from runcard.

        Possibly looking into previous steps outputs.
        Parameters defined in Parameters class are removed from `parameters`
        before `cls` is created.
        Then `nshots` and `relaxation_time` are assigned to cls.

        .. todo::

            move the implementation to History, since it is required to resolve
            the linked outputs

        """
        default_parent_parameters = deepcopy(DEFAULT_PARENT_PARAMETERS)
        parameters = deepcopy(input_parameters)
        for parameter, value in default_parent_parameters.items():
            default_parent_parameters[parameter] = parameters.pop(parameter, value)
        instantiated_class = cls(**parameters)
        for parameter, value in default_parent_parameters.items():
            setattr(instantiated_class, parameter, value)
        return instantiated_class
```

Reply:
With the change in `Parameters.load()` all solutions pass test locally on dummy, so it's just a matter of how the code is self-explainable and robust to future changes. I'll leave it up to you.